DashboardClient/DashboardClient.cpp

```diff
@@ -27,6 +27,160 @@ namespace Umati
+							break;
+						}
+						if(!found) {
+							missingElements.push_back(el);
+						}
+					}
+					//Remove and Unsubscribe the elements
+					for(auto& el : missingElements) {
+						std::shared_ptr<ModelOpcUa::PlaceholderNode> placeholderNodeUnconst = std::const_pointer_cast<ModelOpcUa::PlaceholderNode>(placeholderNode);
+						placeholderNodeUnconst->removeInstance(el);
+						deleteAndUnsubscribeNode(el);
+					}
+				}
+			}
+			//Hotfix clear subscription cache.
+			m_subscribedValues.clear();
+		}
+		void DashboardClient::deleteAndUnsubscribeNode(ModelOpcUa::PlaceholderElement placeHolderElement) {
+			std::shared_ptr<const ModelOpcUa::SimpleNode> element = placeHolderElement.pNode;
+			const std::list<std::shared_ptr<const ModelOpcUa::Node>> children = element->ChildNodes;
+			for(auto& el : children) {
+				std::shared_ptr<const ModelOpcUa::Node> node = el;
+				std::shared_ptr<const ModelOpcUa::SimpleNode> simpleNode = std::dynamic_pointer_cast<const ModelOpcUa::SimpleNode>(node);
+				if(simpleNode != nullptr) {
+					const ModelOpcUa::SimpleNode simpleN = *simpleNode;
+					deleteAndUnsubscribeNode(simpleN);
+				} else {
+					std::shared_ptr<const ModelOpcUa::PlaceholderNode> placeHolderNode = std::dynamic_pointer_cast<const ModelOpcUa::PlaceholderNode>(node);
+					if(placeHolderNode == nullptr) { continue;}
+					std::list<ModelOpcUa::PlaceholderElement> instances = placeHolderNode->getInstances();
+					for(auto& el : instances) {
+						deleteAndUnsubscribeNode(el);
+					}
+				}
+			}
+			browsedNodes.erase(element->NodeId);
+			browsedSimpleNodes.erase(element->NodeId);
+
+		}
+		void DashboardClient::deleteAndUnsubscribeNode(const ModelOpcUa::SimpleNode simpleNode) {
+			const std::list<std::shared_ptr<const ModelOpcUa::Node>> children = simpleNode.ChildNodes;
+			for(auto node : children) {
+				std::shared_ptr<const ModelOpcUa::SimpleNode> simpleNode = std::dynamic_pointer_cast<const ModelOpcUa::SimpleNode>(node);
+				if(simpleNode != nullptr) {
+					const ModelOpcUa::SimpleNode simpleN = *simpleNode;
+					deleteAndUnsubscribeNode(simpleN);
+				} else {
+					std::shared_ptr<const ModelOpcUa::PlaceholderNode> placeHolderNode = std::dynamic_pointer_cast<const ModelOpcUa::PlaceholderNode>(node);
+					if(placeHolderNode != nullptr) {
+						std::list<ModelOpcUa::PlaceholderElement> instances = placeHolderNode->getInstances();
+						for(std::list<ModelOpcUa::PlaceholderElement>::iterator it = instances.begin(); it != instances.end(); it++) {
+							deleteAndUnsubscribeNode(*it);
+						}
+
+					}
+				}
+			}
+			browsedNodes.erase(simpleNode.NodeId);
+			browsedSimpleNodes.erase(simpleNode.NodeId);
+		}
+		void DashboardClient::subscribeEvents() {
+			auto ecbf = [this](IDashboardDataClient::StructureChangeEvent sce) {
+				if(sce.nodeAdded || sce.referenceAdded) {
+					std::thread t([this, sce](){
+						this->updateAddDataSet(sce.refreshNode);
+						this->Publish();
+    				});
+					t.detach();
```

Comment:
How is ensured, that the DashboardClient instance is available until this thread is completed. (e.g. when started before a shutdown of the application)